include/grpc++/completion_queue.h

```diff
@@ -75,10 +76,18 @@ class CompletionQueue {
   explicit CompletionQueue(grpc_completion_queue *take);
   ~CompletionQueue();
 
-  // Blocking read from queue.
-  // Returns true if an event was received, false if the queue is ready
-  // for destruction.
-  bool Next(void **tag, bool *ok);
+  // Tri-state return for AsyncNext: SHUTDOWN, GOT_EVENT, TIMEOUT
+  enum NextStatus {SHUTDOWN, GOT_EVENT, TIMEOUT};
+
+  // Nonblocking (until deadline) read from queue.
+  // Cannot rely on result of tag or ok if return is TIMEOUT
+  NextStatus AsyncNext(void **tag, bool *ok, gpr_timespec deadline);
```

Comment:
https://msdn.microsoft.com/en-us/library/hh874757.aspx confirms that this is a VS12+ thing. If our C++11 standard is based on VS10, then it's a no-go. I realize that that decision would affect some tests, as some tests are already using std::chrono.